autotest/test_canopy_runoff.py

```diff
@@ -0,0 +1,128 @@
+import pathlib as pl
+from datetime import datetime
+
+import numpy as np
+import pytest
+
+from pynhm.base.adapter import adapter_factory
+from pynhm.base.control import Control
+from pynhm.hydrology.PRMSCanopy import PRMSCanopy
+from pynhm.hydrology.PRMSRunoff import PRMSRunoff
+from pynhm.utils.parameters import PrmsParameters
+
+
+@pytest.fixture(scope="function")
+def control(domain):
+    return Control.load(domain["control_file"])
+
+
+@pytest.fixture(scope="function")
+def params(domain):
+    return PrmsParameters.load(domain["param_file"])
+
+
+class TestPRMSCanopyRunoffDomain:
+    def test_init(self, domain, control, params, tmp_path):
+        tmp_path = pl.Path(tmp_path)
+
+        # get the answer data
+
+        comparison_var_names = [
+            "infil",
+            "dprst_stor_hru",
+            "hru_impervstor",
+            "sroff",
+        ]
+        output_dir = domain["prms_output_dir"]
+
+        # Read PRMS output into ans for comparison with pynhm results
+        ans = {}
+        for key in comparison_var_names:
+            nc_pth = output_dir / f"{key}.nc"
+            ans[key] = adapter_factory(nc_pth, variable_name=key)
+
+        # instantiate canopy
+        input_variables = {}
+        for key in PRMSCanopy.get_inputs():
+            nc_pth = output_dir / f"{key}.nc"
+            if "pkwater_equiv" in str(nc_pth):
+                nc_pth = output_dir / "pkwater_equiv_prev.nc"
+            input_variables[key] = nc_pth
+
+        canopy = PRMSCanopy(control=control, params=params, **input_variables)
+
+
+        # instantiate runoff
+        input_variables = {}
+        for key in PRMSRunoff.get_inputs():
+            nc_pth = output_dir / f"{key}.nc"
+            if "soil_moist" in str(nc_pth):
+                nc_pth = output_dir / "soil_moist_prev.nc"
+            input_variables[key] = nc_pth
+        input_variables["net_ppt"] = None
+        input_variables["net_rain"] = None
+        input_variables["net_snow"] = None
+        runoff = PRMSRunoff(control=control, params=params, **input_variables)
+
+        # wire up output from canopy as input to runoff
+        runoff.set_input_to_adapter("net_ppt", adapter_factory(canopy.net_ppt, "net_ppt"))
```

Comment:
I believe the second variable name can be dropped. AdapterOnedarray only passes that name to super().__init__, which is Adapter and it dosent use it.

Reply:
oops, it does use it to set self._variable. but I'm not sure how useful that is.

Reply:
self._variable is only used by AdapterNetcdf so I think variable_name should only be required for that subclass

Reply:
will leave as is for now and rethink the wiring as we connect more boxes.